src/core/ext/filters/client_channel/subchannel_index.cc

```diff
@@ -73,7 +73,8 @@ static grpc_subchannel_key* subchannel_key_copy(grpc_subchannel_key* k) {
 
 int grpc_subchannel_key_compare(const grpc_subchannel_key* a,
                                 const grpc_subchannel_key* b) {
-  if (g_force_creation) return false;
+  // To pretend the keys are different, return a non-zero value.
+  if (g_force_creation) return 1;
```

Comment:
Hmm... It seems like this would actually cause us to forcibly create a new subchannel even in the case where we get a resolver update containing an address for which we already previously had a subchannel.  The net effect would be that every resolver update would cause us to disconnect and reconnect to every backend, which seems really non-optimal.

Why do we even have this feature in the first place?  It might be better to just remove it, since I can't think of any use-case for it.  We can avoid subchannel sharing via channel args, which is a better mechanism, since we can set it at runtime.

Reply:
Ah, okay -- I didn't realize we were depending on this in a test.  Thanks for the pointer.